### DISCUSSION TITLE
SB5.1 - OCI - Build/UCI/Adapter failed

### DISCUSSION THREAD

**Discussion by heungheung** [Sunbird-Ed/Community#368] [Installation] [answered]
w.r.t. Drafted 5.1 doc
  https://ed.sunbird.org/use/installing-ed-platform/release-5.1.0-draft/build
when running Build/UCI/Adapter, got the following error message
```
Downloaded from central: https://repo.maven.apache.org/maven2/com/fasterxml/jackson/module/jackson-modules-base/2.12.5/jackson-modules-base-2.12.5.pom (3.5 kB at 13 kB/s)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:56 min
[INFO] Finished at: 2023-01-24T06:06:13+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project adapter: Could not resolve dependencies for project com.uci:adapter:jar:5.0.0: Failed to collect dependencies at com.uci:utils:jar:5.0.0: Failed to read artifact descriptor for com.uci:utils:jar:5.0.0: Could not transfer artifact com.uci:utils:pom:5.0.0 from/to github-message-rosa (https://maven.pkg.github.com/samagra-comms/message-rosa): Not authorized , ReasonPhrase:Unauthorized. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException

```

**Comment by heungheung** [comment]
any update? this error actually affects 6 different builds: adapter, dao, inbound, orchestrator, outbound and transformer

**Reply by beepdot** [reply]
@ChakshuGautam Can you please provide your inputs in this thread?

**Reply by beepdot** [reply]
Please try running the jobs in this order
```
Build/UCI/Utils
Build/UCI/message-rosa
Build/UCI/dao
Build/UCI/Adapter
Build/UCI/Inbound
Build/UCI/odk
Build/UCI/Outbound
Build/UCI/Orchestrator
Build/UCI/Transformer
Build/UCI/UCI
Build/UCI/uci-transport-socket
```
Official docs will be updated shortly.

**Reply by heungheung** [reply]
@keshavprasadms 
  Build/UCI/uci-transport-socket
does not exists in the jenkins pipeline jobs template
  https://github.com/project-sunbird/sunbird-devops/tree/release-5.1.0/deploy/jenkins/jobs/Build/jobs/UCI/jobs
is it a doc bug?

**Reply by beepdot** [reply]
@heungheung Please ignore that job. I will remove from the docs. This particular job hasn't made to production yet and is still in testing.

**Reply by heungheung** [reply]
Thanks for confirming. Marking "this thread as answer"

**Comment by heungheung** [comment, marked as answer]
Sorry for re-open this thread

Right now, the doc still contains this job for uci-transport-socket while 2 days ago, new jenkins jobs for build, artifactupload and deploy were added in release-5.1.0
https://github.com/project-sunbird/sunbird-devops/commit/5f95fed430c4b616d58549b372a5f7c449cf0131

Can @keshavprasadms or @santhosh-tg reconfirm release-5.1.0 does NOT require uci-transport-socket?

**Reply by beepdot** [reply]
Its not required @heungheung. I sent a PR for adding these jobs since we might need it in future. 
Also I am unable to find the job in docs. I am referring to this docs - https://ed.sunbird.org/use/installing-ed-platform/release-5.1.0-draft/build

**Reply by heungheung** [reply]
got it. Thanks!

You are right. It has been removed from doc 3 days ago.
